### PR TITLE
74 inhibitionactivation values

### PR DESCRIPTION
**Pull request by azywot**
What happened here:
- merge `ECHO` files with  `BMG` files
- calculate `% ACTIVATION`, `% INHIBITION`, `Z-SCORE` for every (non-outlier) compound and control value
- visualizations of the aforemationed figures  
- add tests

Pasting here the visualizations as well as the resultant data frames:

obraz

*Note that the plots are technically barplots with an actual box plot set to invisible as it allowed me to add more `jitter` -> set to max value*
obraz

*I also double checked upon the numbers calculated and earlier csv files received from HTS center.  Everything seems to be well so far, attaching an exemplary snippet below:*

obraz

**Of course, any suggestions are welcome** 🙏🙏

### CODE REVIEW

File: tests/bmg_plate_test.py [hunk 1]
```diff
@@ -35,3 +40,26 @@ def test_plate_class(plate_summary):
...
 def test_outliers(plate_summary):
     assert plate_summary.z_factor == plate_summary.z_factor_no_outliers
+
+
+def test_calculate_activation_inhibition_zscore(df_stats):
+    values = np.array([5, 3, 3])
+    activation, inhibition, z_score = calculate_activation_inhibition_zscore(
+        df_stats.iloc[0], values, mode="all"
+    )
+    print(activation, inhibition, z_score)
```
Comment (Tremirre) [inline tests/bmg_plate_test.py:50]: redundant print in test

File: tests/bmg_plate_test.py [hunk 2]
```diff
@@ -35,3 +40,26 @@ def test_plate_class(plate_summary):
...
+    assert activation[2] == 100.0 and inhibition[1] == 100.0 and z_score[0] == 3.0
+
+
+def test_get_activation_inhibition_zscore_dict(df_stats):
+    values = np.array([[5, 3, 3], [0, 1, 0]])
+    z_dict = get_activation_inhibition_zscore_dict(df_stats, values, modes=dict())
+    print(z_dict)
+    assert z_dict == {
```
Comment (Tremirre) [inline tests/bmg_plate_test.py:58]: redudant print in test

File: src/data/combine.py [hunk 3]
```diff
@@ -45,3 +47,101 @@ def combine_assays(
...
+        if outliers[row, col] != 1
+    ]
+    return pd.DataFrame(records).rename(columns={"Value": column})
+
+
+def get_activation_inhibition_zscore_df(
+    barcode: str, values_dict: dict
+) -> pd.DataFrame:
+    """
+    Get a dataframe with activation and inhibition values.
```
Comment (Tremirre) [inline src/data/combine.py:75]: doc: add endline between docstring and params

File: src/data/bmg_plate.py [hunk 4]
```diff
@@ -176,3 +183,65 @@ def parse_bmg_files_from_dir(dir: str) -> tuple[pd.DataFrame, np.ndarray]:
...
+
+
+def calculate_activation_inhibition_zscore(
+    df_stats: pd.Series,
+    values: np.ndarray,
+    mode: str = "all",
+    without_pos: bool = False,
+) -> tuple[np.ndarray]:
+    """
+    Calculates the activation and inhibition values for each well.
```
Comment (Tremirre) [inline src/data/bmg_plate.py:195]: docs: missing endline between docstring and params

File: src/data/bmg_plate.py [hunk 5]
```diff
@@ -176,3 +183,65 @@ def parse_bmg_files_from_dir(dir: str) -> tuple[pd.DataFrame, np.ndarray]:
...
+    z_score = (values - df_stats["mean_cmpd"]) / df_stats["std_cmpd"]
+
+    return activation, inhibition, z_score
+
+
+def get_activation_inhibition_zscore_dict(
+    df_stats: pd.DataFrame, plate_values: np.ndarray, modes: list[str]
+) -> dict[str, dict[str, float]]:
+    """
+    Calculates activation and inhibition for each compound in the plates.
```
Comment (Tremirre) [inline src/data/bmg_plate.py:229]: docs: missing newline between docstring and params

File: src/data/bmg_plate.py [hunk 6]
```diff
@@ -176,3 +183,65 @@ def parse_bmg_files_from_dir(dir: str) -> tuple[pd.DataFrame, np.ndarray]:
     df = pd.DataFrame(plate_summaries)
     plate_values = np.asarray(plate_values)
     return df, plate_values
+
+
+def calculate_activation_inhibition_zscore(
+    df_stats: pd.Series,
+    values: np.ndarray,
+    mode: str = "all",
```
Comment (Tremirre) [inline src/data/bmg_plate.py:191]: suggestion: maybe create and enum for modes so that there is less room for errors?

**Review by Tremirre — APPROVED** [review]
Nice job!
Pointed out missing newlines in a few places and added one suggestion.

**Review by AndrzejKaj — APPROVED** [review]

File: src/data/bmg_plate.py (hunk 6 above)
Comment (azywot) [reply src/data/bmg_plate.py:191]: Good idea, will add that 👍

**Comment by azywot** [comment]
> A few questions/suggestions regarding the plots:
> 
>     1. Where is CONTROL_POS on the 1st one (% ACTIVATION)?
> 
>     2. Legend for the dashed lined would be nice + I believe it could be gray or sth as it is just helper line to show z-score boundaries.
> 
>     3. You could plot compounds by wells, not rows (as in the slides from HTS) to make the visualization less 'grouped'. But I am not sure what exactly is desired effect here.

1. Good, observation, I actually forgot to describe this one - this is just zoomed in, the whole plot looks like this:
obraz
2. Good point, it's been added, rn it looks like that:
obraz
3.  I wanted to avoid perceiving the x-axis in a numerical way as the values of `% ACTIVATION` and `% INHIBITION` are not correlated with the placement (at least to my best knowledge 🥴), therefore to make a compact comparison it is possible to zoom in/out and the points are jittered. I would leave it for now to have some visualizations ready to be added to the dashboard, meanwhile I'll share these plots to get some feedback 😊

**Pull request merged**